### PR TITLE
Add @osdk/create-app package

### PR DESCRIPTION
**Pull request by tzyl**
Initial code for package for CLI to bootstrap OSDK apps with guided prompts and multiple tempaltes rather than cloning a starter repo and following instructions to update multiple files

### CODE REVIEW

File: packages/create-osdk-app/package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "create-osdk-app",
+  "private": true,
```
Comment (tzyl) [inline packages/create-osdk-app/package.json:3]: Will turn on publishing in a separate step

File: packages/create-osdk-app/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    "clean": "rm -rf lib dist types build tsconfig.tsbuildinfo",
+    "dev:transpile": "tsup --watch",
+    "fix-lint": "eslint . --fix && dprint fmt --config $(find-up dprint.json)",
+    "lint": "eslint . && dprint check  --config $(find-up dprint.json)",
+    "test": "vitest run",
+    "test:watch": "vitest",
+    "transpile": "tsup",
+    "transpileWatch": "tsup --watch",
+    "typecheck": "tsc-absolute --build",
+    "update-snapshots": "UPDATE_SNAPSHOTS=true vitest run"
```
Comment (tzyl) [inline packages/create-osdk-app/package.json:28]: This is the only custom script in the package all others are the same as monorepolint

File: packages/create-osdk-app/src/cli.ts [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+      consola.info(`Reusing existing project directory`);
+    }
+  } else {
+    consola.info(`Creating project directory`);
+    fs.mkdirSync(root, { recursive: true });
+  }
+
+  consola.info(`Copying files into project directory`);
+
+  const templatesDir = findUpSync("templates", {
```
Comment (tzyl) [inline packages/create-osdk-app/src/cli.ts:124]: Originally this had a hardcoded relative path but this is slightly different now running from test or bundle:

When run in vitest this looks for `create-osdk-app/src/cli.ts` -> `create-osdk-app/templates`
When run in tsup bundle this looks for `create-osdk-app/build/js/index.mjs` -> `create-osdk-app/templates`

File: packages/create-osdk-app/package.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "create-osdk-app",
```
Comment (styu) [inline packages/create-osdk-app/package.json:2]: should this have the `@osdk/` prefix? or is that handled somewhere else?

**Review by styu — COMMENTED** [review]
setup largely lgtm. i can't comment on the file directly, but it does feel like we should find a different .svg instead of the resource icon to use?

**Review by mfedderly — COMMENTED** [review]
I'm good with the changes outside of packages/create-osdk-app. I didn't look at the code inside of that package though.

File: packages/create-osdk-app/package.json (hunk 4 above)
Comment (mfedderly) [reply packages/create-osdk-app/package.json:2]: (I think that it has to be named this in order for us to match the other `npx create-foo-app` syntax floating around the JS ecosystem)

One thing we want to do though is make sure the NPM token has permissions to publish to this package as well.

**Pull request merged**